src/main/scala/com/microsoft/ml/spark/codegen/Wrappable.scala

```diff
@@ -0,0 +1,392 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.codegen
+
+import java.lang.reflect.{ParameterizedType, Type}
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+import com.microsoft.ml.spark.core.env.FileUtilities
+import com.microsoft.ml.spark.core.serialize.ComplexParam
+import jdk.jshell.spi.ExecutionControl.NotImplementedException
+import org.apache.spark.ml.evaluation.Evaluator
+import org.apache.spark.ml.{Estimator, Transformer}
+import org.apache.spark.ml.param._
+
+import scala.collection.Iterator.iterate
+import scala.reflect.ClassTag
+
+object CodegenUtils {
+  def indent(test: String, numTabs: Int): String = {
+    test.split("\n".toCharArray).map(l => "    " * numTabs + l).mkString("\n")
+  }
+
+  def camelToSnake(str: String): String = {
```

Comment:
do you assume you can transform variable/classname/method names bi-directionally between python/scala using this conversions? if so maybe add a test to validate? thinking about conflicts.

Reply:
nahh this is just something to make the generated test files a bit more pythonic